src/lxc/start.c

```diff
@@ -341,14 +341,10 @@ static int lxc_serve_state_clients(const char *name,
 	struct state_client *client;
 	struct lxc_msg msg = {.type = lxc_msg_state, .value = state};
 
-	process_lock();
-
-	/* Only set state under process lock held so that we don't cause
-	*  lxc_cmd_add_state_client() to miss a state.
```

Comment:
TBH given the comment here I don't understand why the process_unlock happens before lxc_monitor_send_state().  It seems like the set_state and send_state should happen atomically together.

Reply:
This is the legacy lxc-monitord binary which doesn't use the command socket at
all.